### PR TITLE
Unused elements dashboard; fixes 62

### PR DESCRIPTION
**Pull request by justineatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Add unused and unreferenced elements dashboards.
+ Correct the error in option filter

### CODE REVIEW

File: COMETwebapp/IterationService/IterationService.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+﻿namespace COMETwebapp.IterationService
```
Comment (samatrhea) [inline COMETwebapp/IterationService/IterationService.cs:1]: - missing code header
  - there does not seem to be a unit test for this class, please add

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: COMETwebapp/Pages/ModelDashboard.razor [hunk 2]
```diff
@@ -26,38 +26,55 @@ Copyright (c) 2022 RHEA System S.A.
...
-            @if (ISessionAnchor.OpenIteration.ActualFiniteStateList.Any())
-            {
-                <div class="m-left-10px">
-                    <h6>Filter on states :</h6>
-                    <select @bind="this.FilterState" @oninput="UpdateFilterState">
-                        <option value="">--Select--</option>
-                        @foreach(var stateList in ISessionAnchor.OpenIteration.ActualFiniteStateList)
-                        {
-                            @foreach(var state in stateList.ActualState)
+        <DxTabs @bind-ActiveTabIndex="@ActiveTabIndex">
```
[the remark this answers is not in the document]
Comment (alexatrhea) [reply COMETwebapp/Pages/ModelDashboard.razor:29]: I agree

**Review by alexatrhea — CHANGES_REQUESTED** [review]

File: COMETwebapp/Components/ModelDashboard/UnreferencedElements.razor [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    /// List of unreferenced element defintion from the opened iteration
+    /// </summary>
+    private List<ElementDefinition> UnReferencedElements { get; set; }
+
+    /// <summary>
+    /// Initialize the UnReferencedElements list
+    /// </summary>
+    protected override void OnInitialized()
+    {
+        IterationService iterationService = new IterationService(ISessionAnchor.OpenIteration);
```
Comment (samatrhea) [inline COMETwebapp/Components/ModelDashboard/UnreferencedElements.razor:68]: convention is to use the `var` keyword, so please replace with:

`var iterationService = new IterationService(ISessionAnchor.OpenIteration);`

File: COMETwebapp/Components/ModelDashboard/UnusedElements.razor [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    /// List of unused element defintion from the opened definition
+    /// </summary>
+    private List<ElementDefinition> UnUsedElements { get; set; }
+
+    /// <summary>
+    /// Initialize the UnUsedElements list
+    /// </summary>
+    protected override void OnInitialized()
+    {
+        IterationService iterationService = new IterationService(ISessionAnchor.OpenIteration);
```
Comment (samatrhea) [inline COMETwebapp/Components/ModelDashboard/UnusedElements.razor:68]: convention is to use the `var` keyword, so please replace with:

`var iterationService = new IterationService(ISessionAnchor.OpenIteration);`

File: COMETwebapp.Tests/IterationServiceTest/IterationServiceTest.cs [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            this.iteration.Element.Add(elementDefinition_2);
+            this.iteration.TopElement = elementDefinition_1;
+
+            this.iterationService = new IterationService(this.iteration);
+        }
+
+        [Test]
+        public void VerifyGetParameterValueSets()
+        {
+            Assert.IsNotEmpty(iterationService.GetParameterValueSets());
```
Comment (samatrhea) [inline COMETwebapp.Tests/IterationServiceTest/IterationServiceTest.cs:143]: we typically make use of the `Assert.That` syntax, I think this then becomes:

`Assert.That(iterationService.GetParameterValueSets(), Is.Not.Empty);`

**Review by samatrhea — CHANGES_REQUESTED** [review]

**Review by samatrhea — APPROVED** [review]

**Pull request closed without merging**